src/helpers/MSFS/parseCFGFile.js

```diff
@@ -26,6 +25,123 @@ export function loadCFG(path) {
+ */
+
+export function getHighestPlaneNumber() {
+  let highest = 0;
+  const data = parseCFGFile();
+  Object.keys(data.FLTSIM).forEach(function (key) {
+    key = parseInt(key);
+    if (key > highest) highest = key;
+  });
+
+  return highest;
+}
+
+/**
+ * Add airplane object to CFG FIle
+ *
+ * @export
+ * @argument {Object} airplaneObject Pass the airplane json object to add it to the .CFG file
+ * @argument {Number} planeNumber Plane number for in the aircraft.cfg, only do when overriding!
+ * @example addAirplane({"title":"Airbus A320 Neo Asobo","model":"","panel":"","sound":"","texture":"","kb_checklists":"Boeing747-400_check","kb_reference":"Boeing747-400_ref","description":"TT:AIRCRAFT.DESCRIPTION","wip_indicator":0,"ui_manufacturer":"TT:AIRCRAFT.UI_MANUFACTURER","ui_type":"TT:AIRCRAFT.UI_MODEL","ui_variation":"TT:AIRCRAFT.LIVERY.DEFAULT","ui_typerole":"Commercial Airliner","ui_createdby":"Asobo Studio","ui_thumbnailfile":"","ui_certified_ceiling":"39800","ui_max_range":"3500","ui_autonomy":"7","ui_fuel_burn_rate":"5300","atc_id":"ASX320","atc_id_enable":"0","atc_airline":"","atc_flight_number":"1123","atc_heavy":"1","atc_parking_types":"GATE,RAMP,CARGO","atc_parking_codes":"","atc_id_color":"","atc_id_font":"","isAirTraffic":"0","isUserSelectable":"1"})
+ */
+export function addAirplane(airplaneObject, planeNumber) {
+  planeNumber = planeNumber || getHighestPlaneNumber() + 1;
+  //Check if all values are in here, if not set them to an empty string :)
+  airplaneObject = checkAirplaneObject(airplaneObject);
+  let airplaneCFG = parseCFGFile();
+  airplaneCFG.FLTSIM[planeNumber] = airplaneObject;
+  fs.writeFileSync(CFGpath, ini.stringify(airplaneCFG));
+  return;
+}
+
+/**
+ * Remove airplane object from the CFG file by FLTSIM number
+ *
+ * @export
+ * @argument {Number} planeNumber Plane number for in the aircraft.cfg
+ * @example removeAirplane(63)
+ */
+export function removeAirplane(planeNumber) {
+  let airplaneCFG = parseCFGFile();
+  delete airplaneCFG.FLTSIM[planeNumber];
+  fs.writeFileSync(CFGpath, ini.stringify(airplaneCFG));
+  return;
+}
+
+/**
+ * Check if plane number exists in CFG File
+ *
+ * @export
+ * @argument {Number} planeNumber Plane number to check FLTSIM.number
+ * @returns {Boolean} true:false
+ * @example doesPlaneNumberExist(64)
+ */
+export function doesPlaneNumberExist(planeNumber) {
+  const data = parseCFGFile();
+  return data.FLTSIM[planeNumber] != undefined;
+}
+
+/**
+ * Check if the airplane object is valid, and change when needed
+ *
+ * @export
+ * @argument {Object} airplaneObject Airplane object to validate
+ * @returns {Object} Validated airplaneObject
+ * @example  checkAirplaneObject({"title": "Airbus A320", "model": ""})
+ */
+export function checkAirplaneObject(airplaneObject) {
+  //Check and set values when needed
+  if (airplaneObject.title == undefined || typeof airplaneObject.title != 'string') airplaneObject.title = 'Airbus A320 Neo Asobo';
```

Comment:
The first half of all these `if`s isn't needed (`airplaneObject.title == undefined || `). If the value is undefined, the `typeof` will be `'undefined'`, so the second half will catch it. This will also catch `null`s as their type is `object`.

Removing the first half of the ifs will clean this up a bit!

Another way could be instantiating a new object altogether... For example:

```js
return {
  title: airplaneObject.title || 'Airbus A320 Neo Asobo',
  // ...
};
```

Reply:
Ah thanks! Yeah I never used the typeof stuff before, so it was new for me :D I will use your solution, looks better!

Reply:
Btw, your solution, will this prevent that for example a number is given were a string should be? Since that should be protected

Reply:
Now I think of it, no it won't. Would it be better to just handle that silently by converting it to a string?

Reply:
so title.toString() and for numbers Number(number value)?

Reply:
However number conversion might fail, and then it will give NaN for example right? so that's why I used the if stuff

Reply:
Nvm, however, if a number is 0 for example it would get the value after the || which could be problem

Reply:
Since CFG uses booleans in 0/1 values